src/main/scala/org/economicsl/auctions/sandbox.sc

```diff
@@ -30,4 +32,39 @@ val order7: MarketBidOrder = MarketBidOrder(issuer, Quantity(100), tradable)
 val order8: MarketBidOrder with SingleUnit = MarketBidOrder(issuer, tradable)
 
 // Create a single-unit limit bid order...
-val order9: LimitBidOrder with SingleUnit = LimitBidOrder(issuer, Price(5.5), tradable)
+val order9: LimitBidOrder with SingleUnit = LimitBidOrder(issuer, Price(9.5), tradable)
+
+
+// Create a four-heap order book and add some orders...
+val orderBook = FourHeapOrderBook.empty[LimitAskOrder with SingleUnit, LimitBidOrder with SingleUnit]
+val orderBook2 = orderBook + order3
+val orderBook3 = orderBook2 + order4
+val orderBook4 = orderBook3 + order9
+val orderBook5 = orderBook4 + order8
+
+
+// take a look at paired orders
+val (pairedOrders, _) = orderBook5.takeWhileMatched
+pairedOrders.toList
+
+
+// Implement a weighted average pricing rule...
+case class WeightedAveragePricing(weight: Double) extends DiscriminatoryPricingRule {
+
+  def apply(pair: (LimitAskOrder, LimitBidOrder)): Price = pair match {
+    case (askOrder, bidOrder) => Price((1 - weight) * askOrder.limit.value + weight * bidOrder.limit.value)
+  }
+
```

Comment:
@bherd-rb I think that the pricing rules should be put into the core library.  In most all cases users would want to implement an off-the shelf pricing rule.  We should try to encourage uses implementing bespoke pricing rules to contribute them back...